lib/src/builder/parsing/members/factory.dart

```diff
@@ -56,7 +56,7 @@ class BoilerplateFactory extends BoilerplateMember {
   bool get isFunctionComponentFactory =>
       node.variables.firstInitializer != null &&
       anyDescendantIdentifiers(
-          node.variables.firstInitializer, (identifier) => identifier.name == 'uiFunction');
+          node.variables.firstInitializer, (identifier) => identifier.isFunctionType);
```

Comment:
I don't think we should base whether it's a function component off of a set of known function names, but rather use the presence of a generated config to tell instead.

This getter is also used to determine whether to generate a config, and for the case of `uiJsComponent`, the name `isFunctionComponentFactory` isn't accurate, since it's not a function component.

So, I think this should be something like:
```dart
  bool get shouldGenerateConfig => node.referencesGeneratedConfig;
```

Reply:
Yep, good call. During my first pass I went back and forth here but thought going off the name was still fair (though should definitely have updated `isFunctionComponentFactory` - sorry about that). With fresh eyes though, you're definitely right!